### PR TITLE
Add Registration Gate Tutorial

### PR DESCRIPTION
**Pull request by jobannon**
### Summary 
There seems to be some confusion on how these two gating flows work (registration and email verification).  
Add a second explainer tutorial as well as some common questions asked and troubleshooting tips.

Closes https://github.com/FusionAuth/fusionauth-site/issues/892

### Images
- [x] < `200kb`
- [x] `top-cropped` and `bottom-cropped`
- [x] No RED MAINT MODE

### Content
- [x] `ctrl f . space space` 
- [x] Title Case on Headers
- [x] Breadcrumbs formatted corectly ( `[breadcrumb]#foo -> bar -> baz#`)
** [ ] `grep -n '```' site/_posts/<file>`
- [x] Lists Caps and periods if sentence
- [x] Links fully qualified
- [x] [Note.since] if appropriate

### Perfunctory
- [x] Grammarly review
- [x] Jekyll Errors
- [x] Read Aloud!

### CODE REVIEW

**Review by mooreds — CHANGES_REQUESTED** [review]
Overall pretty good.

* Please run through google docs or grammarly if you haven't (I thought I saw one typo).
* There's a lot of overlap, is there room for a single template that pulls in different pieces? Or is that overkill?
* I think it'd be useful to have the examples all be pied piper/hooli rather than google/youtube
* I think you do a good job of emphasizing that it is rare that folks what registration verification, but I added reasons why you might. That might be good.
* All we have in the API for `user.verified` is a very simple explanation. It'd be good to add in some of the details you outlined in this tutorial to that API doc (which probably will get a lot more eyes).

**Comment by jobannon** [comment]
@mooreds This is ready for merge; I put notes in your initial comment.
@robotdan Let me know if you have any thoughts on this PR.  

Can merge this week if all is good.

File: site/_layouts/doc.liquid [hunk 1]
```diff
@@ -426,6 +426,14 @@
...
+                    <li {% if page.url contains "/gating/" %} class="open"{% endif %}>
+                      <a href="#" class="sub-menu"><i class="fal fa-fw fa-shield-alt"></i>User Control & Gating<i class="fal fa-chevron-{% if page.url contains "/guides/" %}up{% else %}down{% endif %} fa-fw"></i></a>
+                      <ul>
+                        <li {% if page.url == "/docs/v1/tech/tutorials/gate-accounts-until-user-email-verified/" %}class="active"{% endif %}><a href="/docs/v1/tech/tutorials/gating/gate-accounts-until-user-email-verified/">Gate Unverified Users</a></li>
+                        <li {% if page.url == "/docs/v1/tech/tutorials/registration-gate-accounts-until-verified/" %}class="active"{% endif %}><a href="/docs/v1/tech/tutorials/gating/registration-gate-accounts-until-verified/">Gate Unverified Registrations</a></li>
+                        <li {% if page.url == "/docs/v1/tech/tutorials/setting-up-user-account-lockout/" %}class="active"{% endif %}><a href="/docs/v1/tech/tutorials/setting-up-user-account-lockout/">User Account Lockout</a>
```
Comment (mooreds) [inline site/_layouts/doc.liquid:434]: I think you need to remove this from the other section of the menu it is in.

File: site/_layouts/doc.liquid [hunk 2]
```diff
@@ -426,6 +426,14 @@
                   <a href="#" class="sub-menu"><i class="fal fa-fw fa-graduation-cap"></i> Tutorials <i class="fal fa-chevron-{% if page.url contains "/tutorials/" %}up{% else %}down{% endif %} fa-fw"></i></a>
                   <ul>
                     <li {% if page.url == "/docs/v1/tech/tutorials/" %}class="active"{% endif %}><a href="/docs/v1/tech/tutorials/">Overview</a></li>
+                    <li {% if page.url contains "/gating/" %} class="open"{% endif %}>
+                      <a href="#" class="sub-menu"><i class="fal fa-fw fa-shield-alt"></i>User Control & Gating<i class="fal fa-chevron-{% if page.url contains "/guides/" %}up{% else %}down{% endif %} fa-fw"></i></a>
+                      <ul>
+                        <li {% if page.url == "/docs/v1/tech/tutorials/gate-accounts-until-user-email-verified/" %}class="active"{% endif %}><a href="/docs/v1/tech/tutorials/gating/gate-accounts-until-user-email-verified/">Gate Unverified Users</a></li>
```
Comment (mooreds) [inline site/_layouts/doc.liquid:432]: the page.url links here don't include `gating`. Is that intentional?

File: site/docs/v1/tech/tutorials/gating/_common-questions-gating.adoc [hunk 3]
```diff
@@ -0,0 +1,22 @@
+=== Retroactive Emails Once Gate Enabled?
+
+**Question**: Will all my existing users get emails for either gate once enabled?
```
Comment (mooreds) [inline site/docs/v1/tech/tutorials/gating/_common-questions-gating.adoc:3]:
```suggestion
**Question**: Will all my existing users retroactively get emails for either gate once enabled?
```

File: site/docs/v1/tech/tutorials/gating/_common-questions-gating.adoc [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+=== Are Manual Changes Possible?
+
+**Question**: What if I have accidentally/inadvertently imported users with a `verified: false` value (either on the user or application registration)?  In other words, I need all my existing users to have a `verified: true` value.
+
+**Answer**: Currently, there is no way to change this value manually. However, there is an open issue adding this functionality. Please upvote if you believe this is needed.
+
+https://github.com/FusionAuth/fusionauth-issues/issues/1319
+
+=== What About Enabling Both Gates?
```
Comment (mooreds) [inline site/docs/v1/tech/tutorials/gating/_common-questions-gating.adoc:17]:
```suggestion
```

File: site/docs/v1/tech/tutorials/gating/_common-questions-gating.adoc [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+**Question**: What does it mean to enable both registration and email gates and verification paths?
+
+**Answer**: If you enable both registration verification and email verification, there will be a bit of redundancy (this is fine and expected). Consider the use case of ten applications and one common tenant. Maybe nine of those applications require only email verification, but the tenth application requires an application gate to force user opt-in behaviors.
```
Comment (mooreds) [inline site/docs/v1/tech/tutorials/gating/_common-questions-gating.adoc:21]: You don't answer the question directly here. 

Will the user get 2 emails for the 10th application? (I am guessing yes.)

File: site/docs/v1/tech/tutorials/gating/_troubleshooting-gating.adoc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Users cannot verify their application registration or verify their email. The link or code does not work (it is expired or there is some other error).
+
+====  Check the Email Template
+
+Ensure you have selected the correct email template.
+
+**For email verification**
+
+If attempting to verify email, ensure you have selected the appropriate template (`email verification` as opposed to `registration verification` in this instance)
```
Comment (mooreds) [inline site/docs/v1/tech/tutorials/gating/_troubleshooting-gating.adoc:12]: This sentence reads weird. Maybe "If attempting to verify email addresses..."

Comment (mooreds) [reply site/docs/v1/tech/tutorials/gating/_troubleshooting-gating.adoc:12]: Make sure the text matches case of the image: `Email Verification`, etc.

File: site/docs/v1/tech/tutorials/gating/_troubleshooting-gating.adoc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+image::tutorials/email-verify-right.png[Correct Template Email Verification,width=1200]
+image::tutorials/email-verify-wrong.png[Incorrect Template Email Verification,width=1200]
```
Comment (mooreds) [inline site/docs/v1/tech/tutorials/gating/_troubleshooting-gating.adoc:15]: I like the 'checkmark' and 'prohibited' stamps, but the prohibited one should be red.

File: site/docs/v1/tech/tutorials/gating/_troubleshooting-gating.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+If attempting to verify application registration, ensure you have selected the appropriate template (`registration verification` as opposed to `email verification` in this instance)
+
+image::tutorials/registration-verify-right.png[Correct Template Registration,width=1200]
+image::tutorials/registration-verify-wrong.png[Incorrect Template Registration,width=1200]
+
+==== Check External Identifier Timeouts
+
+Ensure your external identifier timeouts are appropriate. Reasonable defaults are set, but they can be adjusted ([breadcrumb]#Tenants -> Advanced Tab -> External identifier durations#), with particular attention paid to `Email verification` and `Registration verification` durations.
+
+image::tutorials/tenant-timeouts.png[Tenant Timeouts Screenshot,width=1200]
```
Comment (mooreds) [inline site/docs/v1/tech/tutorials/gating/_troubleshooting-gating.adoc:28]: use red squares, not black arrows, to outline critical fields.

File: site/docs/v1/tech/tutorials/gating/registration-gate-accounts-until-verified.adoc [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+Once you have verified the email address by clicking on the link, you can now log in and will be sent to the configured redirect URL.
+
+== The User Object
+
+The user object changes before, during, and after the registration verification occurs. Let's look at the JSON.
+
+If a user is created and registration verification is disabled, the verified attribute of the registration object will be true. This is because FusionAuth treats all users as having verified registration (via email) when registration verification is disabled.
+
+[source,json]
+.User Object With Registration Verification Turned Off Initially JSON
```
Comment (mooreds) [inline site/docs/v1/tech/tutorials/gating/registration-gate-accounts-until-verified.adoc:151]: This isn't really the user object at this point. It is the Registrations array, excerpted from the User object.

Please update the title of this JSON fragment.

**Review by mooreds — CHANGES_REQUESTED** [review]
full sentences and run through spell check.

the point of registration verification is that the user who is registered for an application explicitly verified their registration via clicking on a link. So the point is both parties know the user explicitly verified. This is in contrast to user verification, where the point is that the email address is really controlled by the person who signed up for the account. This was a bit confusing to me in the piedpiperchat/notahotdog example (which was otherwise great!).

Is all verification a paid feature or only gating? You mix the two a bit, so it might be worth calling it out. What does gating mean? The user must be verified before they are allowed to login (the JWT is never issued), correct? 

common questions are good but the titles are a bit loud. Maybe include the title in the question? I made those suggested edits.

General menu weirdness
- looks like guide is tagged
- user account lockout in two places
- need redirect if you are moving user account lockout to a different directory (and if it is under a menu, it should be moved to that directory).

Should merge master to this branch

**Comment by mooreds** [comment]
@jobannon is this ready for re-review?

**Pull request merged**